tempodb/backend/raw.go

```
@@ -91,7 +91,10 @@ func (w *writer) CloseAppend(ctx context.Context, tracker AppendTracker) error {
 func (w *writer) WriteTenantIndex(ctx context.Context, tenantID string, meta []*BlockMeta, compactedMeta []*CompactedBlockMeta) error {
 	// If meta and compactedMeta are empty, call delete the tenant index.
 	if len(meta) == 0 && len(compactedMeta) == 0 {
-		return w.w.Delete(ctx, TenantIndexName, KeyPath([]string{tenantID}), false)
+		if err := w.w.Delete(ctx, TenantIndexName, KeyPath([]string{tenantID}), false); err != nil && err != ErrDoesNotExist {
```

maybe more simple?

```
if err := delete(); err == ErrDoesNotExist {
  // eat DNE error on delete
  return nil
}
return err
```

---

Simpler, but doesn't compile.

---

Added a comment about eating the error.